pkg/ruler/rulestore/local/local_test.go

```diff
@@ -83,3 +89,53 @@ func TestClient_LoadAllRuleGroups(t *testing.T) {
 		require.Equal(t, rulespb.ToProto(u, namespace2, ruleGroups.Groups[0]), actual[1])
 	}
 }
+
+func TestListRules_withConfigMapSharding(t *testing.T) {
+	user1 := "user1"
```

Comment:
I don't think there should be any mention of configmaps in the code or the tests. The limitation from k8s should have no bearing on Loki, and instead we should express the intent that all files are loaded recursively.

Given the structure `/base/<user-id>/<any-amount-of-nesting>/file.yaml`, load all the files for each user.